File: server/Models/Keeps.cs
namespace Vault.Models;


public class Keeps : RepoItem<int>{
    public string CreatorId {get; set;}
    public string Name {get; set;}
    public string Description {get; set;}
    public string Img {get; set;}
    public int Views {get; set;}
    public Account Creator {get; set;}
    public int Kept {get; set;}
}

public class VaultedKeep : Keeps{
    public int VaultKeepId {get; set;}
}

File: client/src/models/Keeps.js


export class Keeps{
    constructor(data){
        this.id = data.id
        this.creatorId = data.creatorId
        this.name = data.name
        this.description = data.description
        this.img = data.img
        this.views = data.views
        this.creator = data.creator
        this.kept = data.kept
        this.vaultKeepId = data.vaultKeepId || 0
        this.tagId = data.tagId || 0
    }
}

